container/view.go

```diff
@@ -64,6 +83,8 @@ type ViewDB interface {
 	Save(*Container) error
 	Delete(*Container) error
 
+	GetByPrefix(string) (string, error)
```

Comment:
`filterByNameIDMatches` indeed takes a `View` but it doesn't really need it, it only uses it to call `view.All()`, so in reality it only needs the `[]Snapshot`. 

I did initially put `GetByPrefix` in the `View` but moved it over to the `ViewDB` because it made more sense to put it in there. Also, we can and should make `GetByPrefix` return a `*Container` rather than just the `ID`, with this we can remove the call to `Get` [here](https://github.com/moby/moby/pull/43624/files#diff-6a6b52416980894ed87d4a7d99f7bdef7296c6f9c38c924884c8a574878c3c83R59).

The way I see it, `ViewDB` will get you `*Containers` and you use the `View` to get a snapshot (`Snaphot`) of the containers.

My thinking is to remove the `memory_store.go` and merge it in the `view`.